### ISSUE TITLE
no-negated-condition extra `ignore undefined` option.

### ISSUE THREAD

**Issue by christopher-buss** [sindresorhus/eslint-plugin-unicorn#2238]
### Description

```ts
a ? b : c // fine
a !== undefined : b : c // not fine, requires a === undefined : c : b
```

These two are both checking that a exists, but since we are doing a negated-condition, the rule requires me to swap these around, although it still makes more sense for both to be kept in the original order. I propose an extra option to the no-negated-condition that would allow for evaluating !== undefined as a positive condition, thus making `a !== undefined : b : c` a valid statement, and ignored by the rule.

### Fail

// Ignore

### Pass

```ts
a ? b : c // fine
a !== undefined : b : c
```

**Comment by fregante** [comment]
I agree with the reasoning, but the rule still applies: There's no particular to use `x is not undefined` over `x is undefined`
